src/store/selectors/subjectsPicker.js

```diff
@@ -1,53 +1,19 @@
 const subjectsPicker = ({ primary, lingual }) => {
-  const maxScoreWrapper = {
-    primary: {},
-    lingual: {}
-  }
-
-  let maxPrimaryScore = 0
-  let maxLingualScore = 0
-  let maxPrimarySubject
-  let maxLingualSubject
-  let emptySubject = {
-    name: '',
-    max: {
-      score: 0,
-      formula: ''
-    }
-  }
-
-  maxPrimarySubject = maxLingualSubject = emptySubject
+  const maxScoreWrapper = {}
+  const primaryArray = primary.filter(subject => subject.active)
+  const lingualArray = lingual.filter(subject => subject.active)
 
-  primary.forEach((subject) => {
-    if (subject.max.score > maxPrimaryScore && subject.active) {
-      maxPrimaryScore = subject.max.score
-      maxPrimarySubject = subject
-    }
-  })
-
-  lingual.forEach((subject) => {
-    if (subject.max.score > maxLingualScore && subject.active) {
-      maxLingualScore = subject.max.score
-      maxLingualSubject = subject
-    }
-  })
+  let maxPrimarySubject = _findBestSubject(primaryArray)
+  let maxLingualSubject = _findBestSubject(lingualArray)
 
-  if (maxLingualSubject.max.score * 4 > maxPrimarySubject.max.score) {
-    let maxPrimaryScore2 = maxLingualScore
-    let maxLingualScore2 = 0
-    let maxPrimarySubject2 = maxLingualSubject
-    let maxLingualSubject2 = emptySubject
+  if (maxLingualSubject.max.score * 4 > maxPrimarySubject.max.score) {               // This one is for humane formula
```

Comment:
If you want to put comments add them in separate lines above code, because in this form they are hard to read and maintain (indentation, order, everything will break if you change anything in the lines of code)